docs/sources/writing-guide/shortcodes/index.md

```diff
@@ -143,6 +143,43 @@ This shortcode inserts a lists of links to the page's subpages and includes the
 {{</* section withDescriptions="true"*/>}}
 ```
 
+## `docs/reference` shortcode
+
+The `docs/reference` shortcode offers more flexible linking than the Hugo builtin `relref` shortcode.
+Reference links wrapped in the `docs/reference` shortcode can be defined with multiple destinations.
+The destination used in a rendered page derives from matching path prefixes:
+
+```markdown
+{{%/* docs/reference */%}}
+[<LABEL>]: "<PROJECT PATH PREFIX> -> <REFERENCE>"
+...
+{{%/* /docs/reference */%}}
+```
+
+- _`LABEL`_ matches a link label used in a reference-style link (`[LINK TEXT][LABEL])`).
+  For more information about reference-style links, refer to [Daring Fireball: Markdown Syntax Documentation](https://daringfireball.net/projects/markdown/syntax#link).
+- _`PROJECT PATH PREFIX`_ is matched against the page’s URL path.
+- _`REFERENCE`_ is the parameter that would typically be used in a `relref` shortcode.
+  It can include the special syntax `<SOMETHING VERSION>` which is first looked up in the front matter before falling back to the version inferred from the page's version.
```

Comment:
```suggestion
  It can include the syntax `<SOMETHING VERSION>`, which is first looked up in the front matter before falling back to the version inferred from the page's version.
```
I start to wonder why this is "special", and doing so is not likely important.

Reply:
Good point, it's not really special. I think in my head I was distinguishing the syntax from a general variable substitution. It's very specifically only for version substitution.

Reply:
I think "before falling" --> "before the reference falls back ..." or "before the version falls back ..."? still needs clarification.

Reply:
What about:

> It can include the syntax `<SOMETHING VERSION>`, which is replaced by the value of the front matter `something_version` if present, or to the version inferred from the page.